src/Array2D.luna

```diff
@@ -0,0 +1,235 @@
+import Std.Foreign.C.Value
+import Std.Foreign
+
+import Dataframes.Utils
+import Dataframes.Types
+
+class Size:
+    SizeVal:
+        rows   :: Int
+        columns:: Int
+
+    def getRows: case self of
+        SizeVal rows _: rows
+
+    def getColumns: case self of
+        SizeVal _ columns: columns
+
+class Stride:
+    StrideVal:
+        strideR:: Int
+        strideC:: Int
+
+    def getStrideR: case self of
+        StrideVal strideR _: strideR
+
+    def getStrideC: case self of
+        StrideVal _ strideC: strideC
+
+class Array2D a:
+    Array2D a
+    Array2DVal:
+        size  :: Size
+        stride:: Stride
+        offset:: Int
+        ptr   :: ManagedPointer a
+
+    def _cSym name:
+        lookupSymbol "2d_array" name
+
+    def _cCall name ret args:
+        self . _cSym name . call ret args
+
+    def getPtr: case self of
```

Comment:
Unnecessary getters (looks like Java).

Reply:
Aren't they necessary because of workaroundish `Array2D` constructor? (Otherwise class users would need to pattern match on that implementation detail to get pointer.)